addons/spreadsheet/static/src/data_sources/data_sources.js

```diff
@@ -86,11 +88,39 @@ export class DataSources extends EventBus {
         return id in this._dataSources;
     }
 
+    async notifyWhenPromiseResolves(promise) {
+        this.pendingPromises.add(promise);
+        await promise
+            .then(() => {
+                this.pendingPromises.delete(promise);
+                this.notify();
+            })
+            .catch(() => {
+                this.pendingPromises.delete(promise);
+                this.notify();
+            });
+    }
+
     /**
      * Notify that a data source has been updated. Could be useful to
      * request a re-evaluation.
      */
     notify() {
+        if (this.pendingPromises.size) {
+            if (!this.nextTriggerTimeOutId) {
+                // evaluates at least every 10 seconds, even if there are pending promises
+                // to avoid blocking everything if there is a really long request
+                this.nextTriggerTimeOutId = setTimeout(() => {
+                    this.nextTriggerTimeOutId = undefined;
+                    if (this.pendingPromises.size) {
+                        console.log("this.notify");
```

Comment:
just this time, i get to be the one to point out the console.log :p

Reply:
thanks, fixed :stuck_out_tongue: